movement/watch_faces/complication/stock_stopwatch_face.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 Andreas Nebinger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include "stock_stopwatch_face.h"
+#include "watch.h"
+#include "watch_utility.h"
+#include "watch_rtc.h"
+
+/*
+    This watch face implements the original F-91W stopwatch functionality
+    including counting hundredths of seconds and lap timing. There are two
+    improvements compared to the functionality of the original: 
+    1. When reaching 59:59 the counter does not simply jump back to zero,
+       but keeps track of hours in the upper right hand corner. (Up to 24h)
+    2. Long pressing the light button toggles the led behaviour: it either
+       turns on on each button press or it doesn't.
+*/
+
+
+static uint32_t _ticks;
+static uint32_t _lap_ticks;
+static uint8_t _blink_ticks;
+static uint32_t _old_seconds;
+static uint8_t _old_minutes;
+static uint8_t _hours;
+static bool _colon;
+static bool _is_running;
+
+static void _fast_tick_callback() {
+    _ticks++;
```

Comment:
I was thinking about this, and wondering about 'missed' ticks (which AFAIK will happen now if anything takes more than 1/128th of a second inside a tick). Since the RTC only gives us seconds (I think?), the only way I could think of to handle this is to count the tick offset till the first RTC time change - i.e. read the RTC time in the fast_tick_callback - and on every subsequent RTC time change reset the ticks as appropriate. This means we are out by at most a couple of ticks that occur in the first fraction of a second, and are unlikely to see an interruption. We could also use this to allow entering sleep mode while the stopwatch was on or at least removing the callback while the face wasn't active.

Thoughts? Maybe this is unnecessary complexity (or I'm simply not understanding how ticks work).

Reply:
Hmm, looks like we can actually use one of the timer counters, and simply read out the current value when we want to update the display. This seems like a more straightforward way of figuring out the time.

It looks (maybe) like TC1 is unused, based on a quick look at the code. You can see how TC0 is configured to do 1000hz callbacks in watch_private.c.

Reply:
Your idea of using TC1 for counting sub-second intervals and simply readout values instead of using movement callbacks seems intriguing! My problem is that do not really have an understanding of how this stuff works... ;-) 
But I will take a deeper look and maybe I can figure out how to set up the counter unit and read out values properly, and if it works, this appears to be the cleaner approach. I still don't know if it was a better approach battery-wise, though. 
Will set this PR to draft in the meantime.

Reply:
I'm almost certainly more clueless than you, and have no real experience with embedded stuff; I seem to remember running sleeps meant none of the rtc ticks were executed during the period I was running the sleep, but I may have misinterpreted what was going on.

I also know nothing about the TC stuff other than what I've just been trying to read in the code and the documentaion. For example, see: https://ww1.microchip.com/downloads/en/DeviceDoc/Atmel-42123-SAM-Timer-Counter-(TC)-Driver_ApplicationNote_AT03263.pdf (in particular, tc_get_count_value)

Note that this refers to a function in the older SDK, and unfortunately there doesn't seem to be any helpers in the current one; i.e. you have to read/write to the appropriate places manually, as in watch_private.c. However, if you download the v3 sdk you can see what those library functions do under the hood, and there's also example code.

Reply:
`hri_tccount32_get_COUNT_COUNT_bf` looks like it might be useful.